external/fv3fit/fv3fit/reservoir/domain.py

```diff
@@ -8,3 +16,213 @@ def slice_along_axis(arr: np.ndarray, inds: slice, axis: int = 0):
     sl = [slice(None)] * arr.ndim
     sl[axis] = inds
     return arr[tuple(sl)]
+
+
+class CubedsphereDivider:
+    def __init__(
+        self,
+        tile_layout: Layout,
+        global_dims: Sequence[str],
+        global_extent: Sequence[int],
+    ):
+        """ Class for dividing global cubedsphere data into ranks for preprocessing.
```

Comment:
```suggestion
        """Class for dividing global cubed sphere data into ranks for preprocessing.
```

Reply:
Can you elaborate on what "preprocessing" means? Is the division into ranks needed for a preprocessing step (which one, why?) or primarily for training reasons? Is it divided in preparation for preprocessing or as part of preprocessing?

Reply:
Added more description to the docstring. Effectively the preprocessing chunks it into ranks as well for more efficient loading time and memory usage.